lib/model/nw/schema/NavigationProperty.js

```diff
@@ -33,6 +34,46 @@ class NavigationProperty extends AnnotationTarget {
     Object.defineProperty(this, "toRole", {
       get: () => rawMetadata.$.ToRole,
     });
+
+    Object.defineProperty(this, "association", {
+      get: () => {
+        const association = this.model.resolveModelPath(this.relationship);
+        if (!association) {
+          throw new Error(
+            `Association for navigation property ${this.name} does not exists.`
+          );
+        }
+        return association;
+      },
+    });
+
+    Object.defineProperty(this, "associationEnd", {
```

Comment:
I think the naming should be different because association has two ends with specific roles. Something like 'target' would be more suitable here in navigation property scope imo.